### ISSUE TITLE
PGS sub wrong timing

### ISSUE THREAD

**Issue by Q3CPMA** [mpv-player/mpv#3016]
Hello, some PGS subs basically appears for a very short time when it should last a lot longer for each segment. It's worth noting that ffplay plays it perfectly.
Here's one example of faulty file (I've encountered more than one):
https://d2.maxfile.ro/vphzfrqdmi.mkv

Tested with mpv and ffmpeg masters (using mpv-build).

**Comment by wm4** [comment]
This file looks pretty broken:
```
| + Block group at 289297
|  + Block (track number 2, 1 frame(s), timecode 1.820s = 00:00:01.820) at 289299
|   + Frame with size 22
|  + Block duration: 0.000000ms at 289327
| + Block group at 289330
|  + Block (track number 2, 1 frame(s), timecode 1.816s = 00:00:01.816) at 289332
|   + Frame with size 13
|  + Block duration: 0.000000ms at 289351
| + Block group at 289354
|  + Block (track number 2, 1 frame(s), timecode 1.751s = 00:00:01.751) at 289357
|   + Frame with size 1280
|  + Block duration: 0.000000ms at 290644
| + Block group at 290647
|  + Block (track number 2, 1 frame(s), timecode 1.758s = 00:00:01.758) at 290651
|   + Frame with size 36822
|  + Block duration: 0.000000ms at 327481
| + SimpleBlock (track number 1, 1 frame(s), timecode 1.752s = 00:00:01.752) at 327484
|  + Frame with size 14469
| + Block group at 341960
|  + Block (track number 2, 1 frame(s), timecode 1.758s = 00:00:01.758) at 341962
|   + Frame with size 3
|  + Block duration: 0.000000ms at 341971
| + SimpleBlock (track number 1, 1 frame(s), timecode 1.835s = 00:00:01.835) at 341974
|  + Frame with size 18359
|+ Cluster at 360341
| + Cluster timecode: 8.076s at 360353
| + Block group at 360357
|  + Block (track number 2, 1 frame(s), timecode 8.076s = 00:00:08.076) at 360359
|   + Frame with size 14
|  + Block duration: 0.000000ms at 360379
| + Block group at 360382
|  + Block (track number 2, 1 frame(s), timecode 8.073s = 00:00:08.073) at 360384
|   + Frame with size 13
|  + Block duration: 0.000000ms at 360403
| + Block group at 360406
|  + Block (track number 2, 1 frame(s), timecode 1.820s = 00:00:01.820) at 360408
|   + Frame with size 3
```
Look how the subtitle timestamps make no sense at all. Before and after that are video packets for the time of about 1.7s-2s.

What retarded program wrote this file?

```
| + Muxing application: Lavf56.40.101 at 301
| + Writing application: Lavf56.40.101 at 317
```

Oh look it was ffmpeg!

**Comment by wm4** [comment]
Also I hear the original file was m2ts, so post a cut of that instead. Don't remux.

**Comment by Q3CPMA** [comment]
I tried to cut the m2ts with `dd if=00006.m2ts of=out.m2ts bs=192 skip=1100000 count=130000` since I was told that m2ts packet size is 192, but some error pops out (playing the file is still possible).
https://d2.maxfile.ro/diufuwqjvp.m2ts

It's probably important to say that these english subs weren't on the bluray originally, they were added manually (no precisions about the "how").
Still, there are no comments about this file saying that such problem was encountered.
